fiftyone/operators/executor.py

```diff
@@ -443,11 +446,34 @@ def dataset(self):
         """The :class:`fiftyone.core.dataset.Dataset` being operated on."""
         if self._dataset is not None:
             return self._dataset
-
-        dataset_name = self.request_params.get("dataset_name", None)
-
-        self._dataset = fod.load_dataset(dataset_name)
-        self._dataset.reload()
+        dataset_name = None
+        # since the dataset may have been renamed, if we have the
+        # dataset_id, get the current dataset_name from the db
+        if self.request_params.get("dataset_id", None) is not None:
+            db = foo.get_db_conn()
+            try:
+                oid = ObjectId(self.request_params.get("dataset_id", None))
+                dataset_name = db.datasets.find({"_id": oid}, {"name": 1})[0][
+                    "name"
+                ]
+            except InvalidId:
+                raise ValueError("Invalid dataset_id " % oid)
+            except:
+                raise ValueError("Dataset with id= '%s' does not exist" % oid)
+
+        if not dataset_name:
+            dataset_name = self.request_params.get("dataset_name", None)
+        else:
+            # set the dataset_name in the context to the name resolved from
+            # the db so downstream calls to load_dataset will work
+            self.request_params["dataset_name"] = dataset_name
+
+        if dataset_name:
+            # loading the dataset like this prevents triggering db
+            # migration check and logging last_loaded_at
+            self._dataset = fod.Dataset(
```

Comment:
Why are you preventing migrations/last loaded updates? This should just `load_dataset()` like any other usage of FO does

Reply:
Calling 'load_dataset' leads to excessive/noisy logging and preventing migrations avoids issues of hanging executions due to accidental/automatic db migrations/migration failures happening without direct user interaction or in a separate process.. To execute/queue the operation, presumably the user must have loaded the dataset at some point, which is when the logging/migration should happen.

Reply:
There is nothing special about delegated execution; it should use the same public interface as everything else. Maybe the task has been queued for 1 day and an admin upgraded the deployment in that time and the orchestrator is now running a newer SDK version. The dataset needs to be migrated